code/modules/clothing/under/accessories/accessory.dm

```diff
@@ -33,22 +33,35 @@
 			inv_overlay = image("icon" = icon, "icon_state" = "[tmp_icon_state]_w", dir = SOUTH)
 	if(color)
 		inv_overlay.color = color
+	if(build_from_parts)
+		inv_overlay.cut_overlays()
+		inv_overlay.add_overlay(overlay_image(icon, "[icon_state]_[worn_overlay]", flags=RESET_COLOR)) //add the overlay w/o coloration of the original sprite
 	return inv_overlay
 
 /obj/item/clothing/accessory/proc/get_mob_overlay()
+	var/I
```

Comment:
I think you could rewrite this code so that icon and loc are both variables and just set them in each if, and then make another image variable to stuff these in.
Use human/update_icons.dm as an example. Just replace result_layer with mob_layer and the icons accordingly.

Reply:
The reason this couldn't go into each if is that the proc can be called when there is already a mob_overlay, meaning the ifs don't run, but the build_from_parts if could run, leading to a potential runtime.

Reply:
Sort of did this? Certainly less of a mess now.